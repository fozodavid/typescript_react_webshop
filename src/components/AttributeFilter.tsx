import React, { ChangeEventHandler } from "react";
import './AttributeFilter.css';


interface ICheckBox {
  name: string
  options: string[]
  handler: ChangeEventHandler<HTMLInputElement>
}

const AttributeFilter: React.FC<ICheckBox> = ({ name, options, handler }) => {
  return (
    <div className="form-group">
      <h2>{name}</h2>
      {options.map(option => {
        return (
          <div key={option} className="checkbox">
            <input onChange={handler} name={name} type="checkbox" id={option} />
            <label htmlFor={option} className="form-label">
              {option}
            </label>
          </div>
        )
      })}
    </div>
  );
}

export default AttributeFilter;