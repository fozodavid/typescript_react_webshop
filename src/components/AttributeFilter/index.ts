export { default as AttributeFilter } from './AttributeFilter'