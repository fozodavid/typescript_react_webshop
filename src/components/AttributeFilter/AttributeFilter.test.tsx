import { render, screen } from '@testing-library/react';
import AttributeFilter from './AttributeFilter'

test('renders component', () => {
  const mockName = 'brands';
  const mockOptions = ['ASUS', 'DELL', 'HP', 'LENOVO', 'MSI'];
  const mockHandler = () => {};
  render(<AttributeFilter name={mockName} options={mockOptions} handler={mockHandler} />);
  expect(screen.getByText(`brands`)).toMatchSnapshot('<h2>brands</h2>')
  for (const opt of mockOptions) {
    expect(screen.getByLabelText(opt))
      .toMatchSnapshot(`<label class="form-label" for=${opt}>${opt}</label>`)
  }
});