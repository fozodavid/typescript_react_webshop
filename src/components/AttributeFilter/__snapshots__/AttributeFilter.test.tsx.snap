// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component: <h2>brands</h2> 1`] = `
<h2>
  brands
</h2>
`;

exports[`renders component: <label class="form-label" for=ASUS>ASUS</label> 1`] = `
<input
  id="ASUS"
  name="brands"
  type="checkbox"
/>
`;

exports[`renders component: <label class="form-label" for=DELL>DELL</label> 1`] = `
<input
  id="DELL"
  name="brands"
  type="checkbox"
/>
`;

exports[`renders component: <label class="form-label" for=HP>HP</label> 1`] = `
<input
  id="HP"
  name="brands"
  type="checkbox"
/>
`;

exports[`renders component: <label class="form-label" for=LENOVO>LENOVO</label> 1`] = `
<input
  id="LENOVO"
  name="brands"
  type="checkbox"
/>
`;

exports[`renders component: <label class="form-label" for=MSI>MSI</label> 1`] = `
<input
  id="MSI"
  name="brands"
  type="checkbox"
/>
`;
